storage/txcache/txListForSenderAsSortedMapItem.go

```diff
@@ -0,0 +1,105 @@
+package txcache
+
+import (
+	"math"
+
+	"github.com/ElrondNetwork/elrond-go/core/atomic"
+	"github.com/ElrondNetwork/elrond-go/storage/txcache/maps"
+)
+
+type senderScoreParams struct {
+	count uint64
+	// Size is in bytes
+	size uint64
+	// Fee is in micro ERD
+	fee uint64
+	gas uint64
+	// Price is in micro ERD
+	minGasPrice uint32
+}
+
+// GetKey return the key
+func (listForSender *txListForSender) GetKey() string {
+	return listForSender.sender
+}
+
+func (listForSender *txListForSender) getLastComputedScore() uint32 {
+	return listForSender.lastComputedScore.Get()
+}
+
+// ComputeScore computes the score of the sender, as an integer 0-100
+func (listForSender *txListForSender) ComputeScore() uint32 {
+	score := uint32(listForSender.computeRawScore())
+	listForSender.lastComputedScore.Set(score)
+	return score
+}
+
+func (listForSender *txListForSender) computeRawScore() float64 {
+	fee := listForSender.totalFee.GetUint64()
+	gas := listForSender.totalGas.GetUint64()
+	size := listForSender.totalBytes.GetUint64()
+	count := listForSender.countTx()
+	minGasPrice := listForSender.cacheConfig.MinGasPriceMicroErd
+
+	return computeSenderScore(senderScoreParams{count: count, size: size, fee: fee, gas: gas, minGasPrice: minGasPrice})
+}
+
+// score for a sender is defined as follows:
+//
+//                           (PPUAvg / PPUMin)^3
+// rawScore = ------------------------------------------------
+//            [ln(txCount^2 + 1) + 1] * [ln(txSize^2 + 1) + 1]
+//
+//                              1
+// asymptoticScore = [(------------------) - 0.5] * 2
+//                     1 + exp(-rawScore)
+//
+// For asymptoticScore, see (https://en.wikipedia.org/wiki/Logistic_function)
+//
+// Where:
+//  - PPUAvg: average gas points (fee) per processing unit, in micro ERD
+//  - PPUMin: minimum gas points (fee) per processing unit (given by economics.toml), in micro ERD
+//  - txCount: number of transactions
+//  - txSize: size of transactions, in kB (1000 bytes)
+func computeSenderScore(params senderScoreParams) float64 {
+	allParamsDefined := params.fee > 0 && params.gas > 0 && params.size > 0 && params.count > 0
+	if !allParamsDefined {
+		return 0
+	}
+
+	PPUMin := float64(params.minGasPrice)
+	PPUAvg := float64(params.fee) / float64(params.gas)
+	PPUScore := math.Pow(PPUAvg/PPUMin, 3)
+
+	countPow2 := float64(params.count) * float64(params.count)
+	countScore := math.Log(countPow2+1) + 1
+
+	// We use size in ~kB
+	const bytesInKB = 1000
+	size := float64(params.size) / bytesInKB
```

Comment:
maybe we should think of switching to integer operations instead for optimizing the cycles spent on computing sender scores.

Reply:
Sender of score is recomputed on tx removal and tx arrival. 10 mil computations is less than 0.5 seconds on a simple benchmark (`Benchmark_computeSenderScore`). If we expect 50k add / removals per round, then the score computations should take ~0.0025 seconds.

But we should switch to integer operations nonetheless, I've added a TODO.

**Also, when reading this I've found a bug - we do not recompute the score upon `evictHighNonceTransactions`. I have to fix this.**